flytekit/common/translator.py

```diff
@@ -0,0 +1,301 @@
+from flytekit.common import constants as _common_constants
+from flytekit.common.interface import TypedInterface
+from flytekit.common.launch_plan import SdkLaunchPlan
+from flytekit.common.nodes import SdkNode
+from flytekit.common.tasks.task import SdkTask
+from flytekit.common.workflow import SdkWorkflow
+from flytekit.models import common as _common_models
+from flytekit.models import interface as interface_models
+from flytekit.models import launch_plan as _launch_plan_models
+from flytekit.models import literals as literal_models
+from flytekit.models.common import RawOutputDataConfig
+from flytekit.models.core import identifier as _identifier_model
+from flytekit.models.core import workflow as _core_wf
+from flytekit.models.core import workflow as workflow_model
+from flytekit.models.core.workflow import BranchNode as BranchNodeModel
+
+FlyteLocalEntity = Union[
+    PythonTask,
+    BranchNode,
+    Node,
+    LaunchPlan,
+    Workflow,
+    ReferenceWorkflow,
+    ReferenceTask,
+    ReferenceLaunchPlan,
+    ReferenceEntity,
+]
+FlyteControlPlaneEntity = Union[SdkTask, SdkLaunchPlan, SdkWorkflow, SdkNode, BranchNodeModel]
+
+
+def to_registerable_case(settings: RegistrationSettings, c: _core_wf.IfBlock) -> _core_wf.IfBlock:
+    if c is None:
+        raise ValueError("Cannot convert none cases to registrable")
+    then_node = get_serializable(settings, c.then_node)
+    return _core_wf.IfBlock(condition=c.condition, then_node=then_node)
+
+
+def to_registerable_cases(
+    settings: RegistrationSettings, cases: List[_core_wf.IfBlock]
+) -> Optional[List[_core_wf.IfBlock]]:
+    if cases is None:
+        return None
+    ret_cases = []
+    for c in cases:
+        ret_cases.append(to_registerable_case(settings, c))
+    return ret_cases
+
+
+# Can make this part of the function in the future.
+GLOBAL_CACHE: Dict[FlyteLocalEntity, FlyteControlPlaneEntity] = {}
+
+
+def get_serializable_references(
+    settings: RegistrationSettings, entity: FlyteLocalEntity, fast: bool
+) -> FlyteControlPlaneEntity:
+    if isinstance(entity, ReferenceTask):
+        cp_entity = SdkTask(
+            type="ignore",
+            metadata=TaskMetadata().to_taskmetadata_model(),
+            interface=entity.typed_interface,
+            custom={},
+            container=None,
+        )
+
+    elif isinstance(entity, ReferenceWorkflow):
+        workflow_metadata = WorkflowMetadata(on_failure=WorkflowFailurePolicy.FAIL_IMMEDIATELY)
+
+        cp_entity = SdkWorkflow(
+            nodes=[],  # Fake an empty list for nodes,
+            id=entity.reference.id,
+            metadata=workflow_metadata,
+            metadata_defaults=workflow_model.WorkflowMetadataDefaults(),
+            interface=entity.typed_interface,
+            output_bindings=[],
+        )
+
+    elif isinstance(entity, ReferenceLaunchPlan):
+        wf_id = _identifier_model.Identifier(_identifier_model.ResourceType.WORKFLOW, "", "", "", "")
```

Comment:
not related to this PR but why do you use an empty identiifer rather than filling out the fields?

Reply:
No intelligent reason

Reply:
this will all get deleted i hope